### PR TITLE
Fedora packaging changes

### PR DESCRIPTION
**Pull request by cmcantalupo**
Several changes are required for fedora packaging

- Build from files committed to git only, do not download binaries
  - Do not build gtest or gmock, use packages
  - Do not download fuzz corpus unless fuzz testing is enabled
  - Create VERSION files in spec file rather than rely on generated versions in archive
- Run unit tests in %check sections
- Relates to #3505

### CODE REVIEW

File: geopmpy/geopmpy.spec.in [hunk 1]
```diff
@@ -46,9 +47,9 @@ BuildRequires: marshalparser
 Python3 package for GEOPM runtime.  Provides the implementation for
 the runtime launcher and post-processing tools.
 
-# Steps for all packages
 %prep
-%setup -n geopmpy-%{version}
+%setup
```
Comment (cmcantalupo) [inline geopmpy/geopmpy.spec.in]:
```suggestion
%setup -n geopm-%{version}
```

File: release/fedora/geopmdpy.spec [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+# Steps for all packages
+%prep
+%setup -n geopm-%{version}
+%patch0 -p1
+cd geopmdpy
+echo %{version} > geopmdpy/VERSION
+
+%build
+cd geopmdpy
+%py3_build
```
Comment (bgeltz) [inline release/fedora/geopmdpy.spec:57]: SETUPTOOLS_SCM_PRETEND_VERSION may help here
https://github.com/pypa/setuptools_scm/blob/main/docs/overrides.md

File: libgeopmd/Makefile.am [hunk 3]
```diff
@@ -348,13 +346,8 @@ io.github.geopm.xml:
...
 # RPM TARGET
-rpm_topdir ?= $(HOME)/rpmbuild
-rpm: dist
-	mkdir -p $(rpm_topdir)/SOURCES
-	mkdir -p $(rpm_topdir)/SPECS
-	cp geopm-service-$(VERSION).tar.gz $(rpm_topdir)/SOURCES
-	cp geopm-service.spec $(rpm_topdir)/SPECS
-	rpmbuild $(rpmbuild_flags) -ba $(rpm_topdir)/SPECS/geopm-service.spec
+rpm:
+	./make_rpm.sh
```
Comment (bgeltz) [inline libgeopmd/Makefile.am]: Double check that defining rpmbuild_flags with make rpm passes the flags through properly for things like levelzero enablement, etc.

A deprecation warning would also be nice.

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: geopmdpy/test/TestActiveSessions.py [hunk 4]
```diff
@@ -202,7 +202,7 @@ def check_json_file(self, name, contents, is_valid):
             mock_srf.assert_called_once_with(full_file_path)
 
             if is_valid:
-                mock_get_session_path.called_once_with(contents['client_pid'])
```
Comment (bgeltz) [inline geopmdpy/test/TestActiveSessions.py:205]: A tech debt issue or similar should be filed about why this code path wasn't being hit.

File: libgeopm/test/Makefile.mk [hunk 5]
```diff
@@ -209,4 +206,13 @@ coverage: init-coverage check
...
 	genhtml coverage-base-combined-filtered.info --output-directory coverage-base --legend -t $(VERSION) -f
 
+if BUILD_GTEST
 include test/googletest.mk
+test_geopm_test_LDADD += libgtest.a \
+                         libgmock.a \
+                         # end
+else
+test_geopm_test_LDADD += -lgmock \
+                         -lgtest \
```
Comment (bgeltz) [inline libgeopm/test/Makefile.mk:216]: If we're going to try to link against gtest and gmock when BUILD_GTEST is disabled, the documentation should be updated to reflect this new requirement.

**Review by bgeltz — CHANGES_REQUESTED** [review]

**Review by bgeltz — APPROVED** [review]

**Pull request merged**